### ISSUE TITLE
Unexpected OAuth.urlDecode behavior

### ISSUE THREAD

**Issue by scyclops** [bytespider/jsOAuth#45]
I was calling an OAuth REST API endpoint with a url that contained a GET parameter with spaces in it.  The url encoder I was using encoded the spaces as plus symbols (+).  When I passed the encoded url (eg. https://domain.com/docs?query=foo+bar ) to the getJSON method it decoded it in an unexpected way such that the plus symbol was left in place instead of being decoded to a space.  Then, when the url was re-encoded before being request was made, the plus symbol was encoded as %2B which caused the server to think the character was a plus symbol instead of a space as was originally intended.  

My solution to this was to use a different encoder that encoded spaces as %20 instead of plus symbols before passing the url to the OAuth library but I was wondering if this was intended behavior or accidental on the part of the OAuth library.

TL;DR

What happened:
OAuth.urlDecode('a+b') === 'a+b'

What I expected:
OAuth.urlDecode('a+b') === 'a b'
